share/commands/agent-check

```diff
@@ -224,24 +227,76 @@ check_and_fix_inputs() {
 }
 
 check_varspace() {
-  # check max space available for databases and stop if there is a risk
-  if [ -d /var/rudder/ldap/ ]; then
-    ldap_space=$(df /var/rudder/ | tail -n 1 | awk '{print $5}' | sed 's/%//')
-    if [ "${ldap_space}" -gt 98 ]; then
-      stop_now=1
+  # check max space available for databases and stop Rudder if there is a risk
+  if [ -f ${RUDDER_SERVER_ROLES}/rudder-ldap ]; then
+    if [ -d /var/rudder/ldap/ ]; then
+      ldap_space=$(df /var/rudder/ldap | tail -n 1 | awk '{print $5}' | sed 's/%//')
+      if [ "${ldap_space}" -gt 98 ]; then
+        stop_now=1
+      fi
     fi
   fi
-  if [ -d /var/lib/postgresql/ ]; then
-    psql_space=$(df /var/lib/postgresql/ | tail -n 1 | awk '{print $5}' | sed 's/%//')
-    if [ "${psql_space}" -gt 98 ]; then
-      stop_now=1
+
+  if [ -f ${RUDDER_SERVER_ROLES}/rudder-reports ]; then
+    # Get the run path of postgresql
+    PG_TABLES_PATH=$(su - postgres -c 'psql -t -c  "show data_directory;"')
+    if [ -z "${PG_TABLES_PATH}" ]; then
+      echo "FATAL: Cannot connect to PostgreSQL - please check its status" | logger -s
+    else
+      psql_space=$(df ${PG_TABLES_PATH} | tail -n 1 | awk '{print $5}' | sed 's/%//')
+
+      # If we are using TCP, we cannot stop postgresql otherwise all infra can go down - but we can stop rudder-jetty and disable rudder agent
```

Comment:
In this case, postgresql will break AND the infra will go down.
hopefully we won't have this tcp problem in 5.1